### PR TITLE
[9.3] Take over #12294: Provide special member functions for TriaAccessor

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

**Review by bangerth — APPROVED** [review]

**Comment by bangerth** [comment]
Is this a duplicate of #12294?

**Comment by bangerth** [comment]
Oh, I see now, it's on the release branch.

**Review by marcfehling — APPROVED** [review]

File: include/deal.II/grid/tria_accessor.h [hunk 1]
```diff
@@ -720,6 +720,17 @@ class TriaAccessor : public TriaAccessorBase<structdim, dim, spacedim>
                const int                           index      = -1,
                const AccessorData *                local_data = nullptr);
 
+  /**
+   * The copy constructor is not deleted but copied constructed elements should
+   * not be modified, also the comments to the copy assignemtn operator.
```
Comment (Rombur) [inline include/deal.II/grid/tria_accessor.h:725]:
```suggestion
   * not be modified, also the comments to the copy assignment operator.
```

**Pull request merged**